test/gtest/intervals.cpp

```diff
@@ -0,0 +1,95 @@
+#include "support/intervals.h"
+#include "ir/utils.h"
+#include "gtest/gtest.h"
+
+using namespace wasm;
+
+TEST(IntervalsTest, TestEmpty) {
+  std::vector<Interval> intervals;
+  std::vector<int> expected;
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals), expected);
+}
+
+TEST(IntervalsTest, TestOne) {
+  std::vector<Interval> intervals;
+  intervals.emplace_back(Interval{0, 2, 5});
+  std::vector<int> expected{0};
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals), expected);
+}
+
+TEST(IntervalsTest, TestNoOverlapFound) {
+  std::vector<Interval> intervals;
+  intervals.emplace_back(Interval{0, 4, 2});
+  intervals.emplace_back(Interval{4, 8, 2});
+  std::vector<int> expected{0, 1};
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals), expected);
+}
+
+TEST(IntervalsTest, TestOverlapFound) {
+  std::vector<Interval> intervals;
+  intervals.emplace_back(Interval{0, 2, 5});
+  intervals.emplace_back(Interval{1, 4, 10});
+  intervals.emplace_back(Interval{4, 5, 15});
+  std::vector<int> expected{1, 2};
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals), expected);
+}
+
+TEST(IntervalsTest, TestOverlapAscendingSequence) {
+  std::vector<Interval> intervals;
+  intervals.emplace_back(Interval{0, 3, 6});
+  intervals.emplace_back(Interval{2, 6, 2});
+  intervals.emplace_back(Interval{3, 15, 5});
+  intervals.emplace_back(Interval{4, 11, 1});
+  intervals.emplace_back(Interval{6, 20, 15});
+  intervals.emplace_back(Interval{12, 18, 3});
+  intervals.emplace_back(Interval{14, 21, 5});
+  intervals.emplace_back(Interval{23, 28, 5});
+  std::vector<int> expected{4, 3, 7};
```

Comment:
Am I reading correctly that this is keeping `{6, 20, 15}` at index 4 followed by `{4, 11, 1}` at index 3? That doesn't seem right.

Reply:
Indeed this expected vector was wrong and the test was failing. it is keeping {0, 4, 7}.